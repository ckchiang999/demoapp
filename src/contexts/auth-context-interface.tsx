import { User } from "../models/authentication/user";

export interface IAuthContext {
  login(username: string, password: string): Promise<boolean>;
  logout(): Promise<void>;
  currentUser(): User | null;
}
