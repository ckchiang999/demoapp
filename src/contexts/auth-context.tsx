import React from "react";
import { IAuthContext } from "./auth-context-interface";

export const AuthContext = React.createContext<IAuthContext | null>(null);
