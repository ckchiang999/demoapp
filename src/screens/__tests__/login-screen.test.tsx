import "react-native";
import React from "react";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react-native";
import { AuthContext } from "../../contexts/auth-context";
import { IAuthContext } from "../../contexts/auth-context-interface";

import LoginScreen from "../login-screen";

describe("LoginScreen", () => {
  it("should render", () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    // Compare to the snapshot of the rendered output
    expect(tree).toMatchSnapshot();
  });

  it("should contain a login button", async () => {
    const authContext = {
      login: (_username: string, _password: string) => {
        return Promise.resolve(true);
      },
      logout: () => {
        return Promise.resolve();
      },
      currentUser: () => {
        return null;
      },
    } as IAuthContext;

    const authContextSpy = jest.spyOn(authContext, "login");

    const component = (
      <AuthContext.Provider value={authContext}>
        <LoginScreen />
      </AuthContext.Provider>
    );
    const renderApi = render(component);
    const title = await renderApi.findByText("Login");
    const loginButton = await renderApi.findByTestId("LoginButton");

    expect(title).toBeTruthy();
    expect(loginButton).toBeTruthy();

    fireEvent(loginButton, "press");
    expect(authContextSpy).toHaveBeenCalled();
  });
});
