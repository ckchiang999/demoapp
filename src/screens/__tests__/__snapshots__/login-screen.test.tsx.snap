// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "elevation": 5,
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingTop": "50%",
    }
  }
>
  <ActivityIndicator
    animating={false}
    color="#BA55D3"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "600",
        "marginBottom": 50,
        "marginTop": 50,
      }
    }
  >
    Login
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    editable={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Username"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#E3E3E3",
        "borderWidth": 1,
        "fontSize": 20,
        "margin": 5,
        "padding": 5,
        "textAlign": "left",
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCompleteType="password"
    editable={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "borderColor": "#E3E3E3",
        "borderWidth": 1,
        "fontSize": 20,
        "margin": 5,
        "padding": 5,
        "textAlign": "left",
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#007AFF",
        "borderRadius": 20,
        "color": "#FFFFFF",
        "margin": 10,
        "padding": 5,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
      testID="LoginButton"
    >
      LOGIN
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "red",
      }
    }
  >
    
  </Text>
</View>
`;
